packages/core-mobile/app/screens/send/components/SendTokenForm.tsx

```diff
@@ -0,0 +1,200 @@
+import AddressBookLists, {
+  AddressBookSource
+} from 'components/addressBook/AddressBookLists'
+import UniversalTokenSelector from 'components/UniversalTokenSelector'
+import { useSendContext } from 'contexts/SendContext'
+import { useAddressBookLists } from 'components/addressBook/useAddressBookLists'
+import FlexSpacer from 'components/FlexSpacer'
+import { Contact, CorePrimaryAccount } from '@avalabs/types'
+import { AddrBookItemType } from 'store/addressBook'
+import AnalyticsService from 'services/analytics/AnalyticsService'
+import { Network, NetworkVMType } from '@avalabs/core-chains-sdk'
+import { TokenWithBalance } from '@avalabs/vm-module-types'
+import { Amount } from 'types'
+
+const SendTokenForm = ({
+  network,
+  maxAmount,
+  addressPlaceholder,
+  error,
+  isValid,
+  isValidating,
+  isSending,
+  onOpenQRScanner,
+  onOpenAddressBook,
+  onSelectContact,
+  onSend
+}: {
+  network: Network
+  maxAmount: Amount | undefined
+  addressPlaceholder: string
+  error: string | undefined
+  isValid: boolean
+  isValidating: boolean
+  isSending: boolean
+  onOpenQRScanner: () => void
+  onOpenAddressBook: () => void
+  onSelectContact: (item: Contact | CorePrimaryAccount) => void
+  onSend: () => void
+}): JSX.Element => {
+  const { setToken, token, setAmount, amount, toAccount } = useSendContext()
+  const [isAddressTouched, setIsAddressTouched] = useState(false)
+  const [isTokenTouched, setIsTokenTouched] = useState(false)
+  const [isAmountTouched, setIsAmountTouched] = useState(false)
+
+  const {
+    showAddressBook,
+    setShowAddressBook,
+    onContactSelected,
+    saveRecentContact,
+    reset: resetAddressBookList
+  } = useAddressBookLists()
+
+  const handleContactSelected = (
+    item: Contact | CorePrimaryAccount,
+    type: AddrBookItemType,
+    source: AddressBookSource
+  ): void => {
+    onSelectContact(item)
+    onContactSelected(item, type)
+    AnalyticsService.capture('SendContactSelected', { contactSource: source })
+  }
+
+  const handleSend = (): void => {
+    saveRecentContact()
+    onSend()
+  }
+
+  const handleMax = useCallback(() => {
+    if (maxAmount !== undefined) {
+      setAmount(maxAmount)
+    }
+  }, [maxAmount, setAmount])
+
+  useEffect(() => {
+    if (toAccount.address) {
+      setShowAddressBook(false)
+    }
+  }, [setShowAddressBook, toAccount.address])
+
+  useEffect(() => {
+    if (!isAddressTouched && toAccount.address) {
+      setIsAddressTouched(true)
+    }
+  }, [toAccount.address, isAddressTouched])
+
+  useEffect(() => {
+    if (!isTokenTouched && token) {
+      setIsTokenTouched(true)
+    }
+  }, [token, isTokenTouched])
+
+  useEffect(() => {
+    if (!isAmountTouched && amount) {
+      setIsAmountTouched(true)
+    }
+  }, [amount, isAmountTouched])
+
+  const isAllFieldsTouched = useMemo(
+    () => isAddressTouched && isTokenTouched && isAmountTouched,
+    [isAddressTouched, isTokenTouched, isAmountTouched]
+  )
```

Comment:
probably easier to use react form or something here that way we don't have to manually manage this logic

Reply:
I tried to use 'react-hook-form' but it didn't fit well here because `<UniversalTokenSelector />` is wrapping two form fields(token and amount) at once. `react-hook-form` assumes that each `<Controller />` render only one form field. Maybe we can try again after separating amount and token inputs to the separate component, but I guess not right now.